news/migrations/0002_remove_news_alert.py

```diff
@@ -0,0 +1,17 @@
+# Generated by Django 2.2.12 on 2020-06-23 10:31
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('news', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
```

Comment:
Should all the old news which are alerts be marked as pinned? Or should we handle alerts someway in the migration?

Reply:
Currently, if the news item has the "alert-field" set but "pin" is false, the appearance on the course front page is not changed. It was actually already referred to earlier: 

> 1. When the user creates a new "news" element, there is a selector called Alert, which asks for the Alert style. However, the selected style is not applied to the unpinned items. Do you think we should apply the class also to unpinned news? or maybe remove the selector when the "Pin" checkbox is unselected?
>    ![image](https://user-images.githubusercontent.com/10437475/84897689-b377ea80-b0ae-11ea-8218-0e90d9ebc094.png)

So I guess those that actually are wanted to show as more noticeable for students via the current alert, already have that "pin" luckily in place :) 

Maybe we could check the database how many courses have news that would be affected by this? I think that would help in deciding if any further actions are needed, if the "pin" is already in place for those that actually use the alerts.

Reply:
> So I guess those that actually are wanted to show as more noticeable for students via the current alert, already have that "pin" luckily in place :)

Makes sense, thus this should be fine. I forgot/missed that part earlier, sorry.

Currently pinned + alerted news are:

```
test/vartiah4/
    2017-03-06 12:44:12+00:00 Diibadaaba
a1141/2018/
    2019-09-03 06:04:18+00:00 Kurssi päättyy
a1141/2019/
    2020-06-08 10:25:35+00:00 Kurssi on päättynyt
a1141/2020
    2020-06-08 10:26:33+00:00 Tietorakenteet ja algoritmit Y
o1/2016/
    2016-12-21 10:16:04+00:00 Epävirallinen suoritustodistus
o1/2017/
    2018-08-24 10:53:30+00:00 Tämä on vuoden 2017 kurssimateriaali / This is an old copy of the course material
    2018-01-09 16:55:03+00:00 Epävirallinen suoritustodistus
o1/2018/
    2019-09-09 10:57:09+00:00 Tämä on vanha versio kurssimateriaalista / This is an old copy of the course material
tietotekniikka-sovelluksissa/2020/
    2020-03-15 17:30:21+00:00 Harjoitustilaisuudet 16.3. alkaen
wsd/2017-2018/
    2018-03-12 10:25:16+00:00 Lecturer is on sick leave - Team Demos on Monday and Tuesday have to be rescheduled
    2018-03-08 08:02:07+00:00 Project demo times available for reservation
```

I think all of these are fine as pinned, which they are when alert is dropped. Looks good :+1:

Reply:
Not many courses are affected by the removal of the alerts, but in any case, those courses will stop working because Django won't find the field `alert `in the table news. am I right? or how are those courses going to respond?

I understand that once this PR is merged those courses will be "past courses", and not many people will use them, but I wonder if we want to have those broken courses around.

Reply:
> those courses will stop working because Django won't find the field alert in the table news. am I right? or how are those courses going to respond?

Incorrect, the "alert" news will be just pinned news for them. The migration just drops the field but the code changes at the same time. all courses use the same django code.